src/components/PublishTesterComponent.vue

```diff
@@ -0,0 +1,171 @@
+<script setup lang="ts">
+import { createPublisher } from '@/lib/roslibUtils/createPublisher';
+import { createSubscriber } from '@/lib/roslibUtils/createSubscriber';
+import { ref, computed } from 'vue';
+import type { TopicType } from '@/lib/roslibUtils/rosTypes';
+
+// Define the message types
+const messageTypes = [
+  'std_msgs/Int32',
+  'std_msgs/Bool',
+  'std_msgs/String',
+  'std_msgs/Char',
+  'std_msgs/Float32',
+  'std_msgs/Time',
+  'sensor_msgs/msg/CompressedImage',
+];
+
+const input1 = ref('');
+const input2 = ref('');
+const input3 = ref('');
+const customMessageType = ref('');
+const receivedMessage = ref('');
+
+const currentConnectionIdx = computed(() => {
+  let idx = messageTypes.indexOf(input2.value);
+  if (idx === -1) idx = messageTypes.length; // Custom input index
+  return idx;
+});
+
+function convertMessage(topicType: string, message: string) {
+  switch (topicType) {
+    case 'std_msgs/Int32':
+      return { data: parseInt(message) };
+    case 'std_msgs/Bool':
+      return { data: message.toLowerCase() === 'true' };
+    case 'std_msgs/String':
+      return { data: message };
+    case 'std_msgs/Char':
+      return { data: message.charAt(0) };
+    case 'std_msgs/Float32':
+      return { data: parseFloat(message) };
+    case 'std_msgs/Time':
+      return { data: new Date(message).toISOString() };
+    case 'sensor_msgs/msg/CompressedImage':
+      return { data: message }; // Assuming the message is a base64 encoded string
+    default:
+      return { data: message };
+  }
+}
+
+function publishTest() {
+  const topicName = input1.value;
+  const topicType = currentConnectionIdx.value === messageTypes.length ? customMessageType.value : input2.value;
+  const topicMessage = convertMessage(topicType, input3.value);
+
+  if (topicName && topicType && topicMessage) {
+    const testPublisher = createPublisher({
+      topicName: topicName,
+      topicType: topicType as TopicType,
+    });
+
+    testPublisher.publish(topicMessage);
```

Comment:
Move this .publish function call so it's after testSubscriber.start as there could be a case where the subscriber does not pick it up due to which one finishes running